platform/security/security-userpass-realm/src/main/java/org/codice/ddf/security/userpass/realm/UsernamePasswordRealm.java

```diff
@@ -0,0 +1,217 @@
+      return simpleAuthenticationInfo;
+    }
+    throw new AuthenticationException("Login failed for user: " + user);
+  }
+
+  private SimplePrincipalCollection createPrincipalCollectionFromSubject(Subject subject) {
+    SimplePrincipalCollection principals = new SimplePrincipalCollection();
+    DefaultSecurityAssertionBuilder assertionBuilder = new DefaultSecurityAssertionBuilder();
+
+    AttributeStatement attributeStatement = new AttributeStatementDefault();
+    Principal userPrincipal =
+        subject
+            .getPrincipals()
+            .stream()
+            .filter(p -> p instanceof UserPrincipal)
+            .findFirst()
+            .orElseThrow(AuthenticationException::new);
+    Set<Principal> rolePrincipals =
+        subject
+            .getPrincipals()
+            .stream()
+            .filter(p -> p instanceof RolePrincipal)
+            .collect(Collectors.toSet());
+    for (ClaimsHandler claimsHandler : claimsHandlers) {
+      ClaimsCollection claims =
+          claimsHandler.retrieveClaims(
+              new ClaimsParametersImpl(userPrincipal, rolePrincipals, new HashMap<>()));
+      mergeClaimsToAttributes(attributeStatement, claims);
+    }
+
+    assertionBuilder
+        .addAttributeStatement(attributeStatement)
+        .addPrincipal(userPrincipal)
+        .weight(SecurityAssertion.LOCAL_AUTH_WEIGHT)
+        .issuer("DDF")
+        .notBefore(new Date())
+        .notOnOrAfter(new Date(new Date().getTime() + 14400000L));
+
+    for (Principal principal : rolePrincipals) {
+      assertionBuilder.addPrincipal(principal);
+    }
+
+    SecurityAssertion assertion = assertionBuilder.build();
+
+    principals.add(assertion, "UP");
+
+    return principals;
+  }
+
+  private void mergeClaimsToAttributes(
+      AttributeStatement attributeStatement, ClaimsCollection claims) {
+    for (Claim claim : claims) {
+      Attribute newAttr = new AttributeDefault();
+      newAttr.setName(claim.getName());
+      newAttr.setValues(claim.getValues());
+      boolean found = false;
+      for (Attribute attribute : attributeStatement.getAttributes()) {
+        if (attribute.getName().equals(newAttr.getName())) {
+          found = true;
+          for (String value : newAttr.getValues()) {
+            attribute.addValue(value);
+          }
+        }
+      }
+      if (!found) {
+        attributeStatement.addAttribute(newAttr);
+      }
+    }
+  }
+
+  public void addRealm(ServiceReference<JaasRealm> serviceReference) {
+    Bundle bundle = FrameworkUtil.getBundle(UsernamePasswordRealm.class);
+    if (null != bundle) {
+      JaasRealm realm = bundle.getBundleContext().getService(serviceReference);
+      LOGGER.trace("Adding validator for JaasRealm {}", realm.getName());
+      realmList.add(realm);
+    }
+  }
+
+  public void removeRealm(ServiceReference<JaasRealm> serviceReference) {
+    Bundle bundle = FrameworkUtil.getBundle(UsernamePasswordRealm.class);
```

Comment:
❗️ See comment in the blueprint with regard to getting rid of this OSGi coupling.